server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -166,6 +170,23 @@ public SourceFieldMapper build() {
             }
             if (isDefault()) {
                 return indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT : DEFAULT;
+            } else if (supportsNonDefaultParameterValues == false) {
+                List<String> disallowed = new ArrayList<>();
+                if (enabled.get().value() == false) {
+                    disallowed.add("enabled");
+                }
+                if (includes.get().isEmpty() == false) {
+                    disallowed.add("includes");
+                }
+                if (excludes.get().isEmpty() == false) {
+                    disallowed.add("excludes");
+                }
+                assert disallowed.isEmpty() == false;
```

Comment:
I think `mode` option `disabled` should be disallowed as well?

Reply:
Indeed, added to the check for default mapper too.